examples/aiml/recog.scm

```diff
@@ -0,0 +1,63 @@
+
+; Some very crude experiments
+;
+; This is a rough sketch of the idea that pattern recognition
+; is the dual of pattern matching.  There are many things wrong
+; with the below; its just a sketch.
+
+; Two different pseudo-AIML rules:
+;    I * you   --> I * you too
+;    I love *  --> I like * a lot!
+;
+(BindLink
+	(ListLink
+		(ConceptNode "I")
+		(VariableNode "$star")
+		(ConceptNode "you"))
+	(ListLink
+		(ConceptNode "I")
+		(VariableNode "$star")
+		(ConceptNode "you")
+		(ConceptNode "too")))
+
+(BindLink
+	(ListLink
+		(ConceptNode "I")
+		(ConceptNode "love")
+		(VariableNode "$star"))
+	(ListLink
+		(ConceptNode "I")
+		(ConceptNode "like")
+		(VariableNode "$star")
+		(ConceptNode "a")
+		(ConceptNode "lot!")))
+
+(define data
+	;; A pretend "sentence" that is the "input".
+	(PatternLink
+		(ListLink
+			(ConceptNode "I")
+			(ConceptNode "love")
+			(ConceptNode "you"))))
+
+;; Perform the search.
+(cog-recognize data)
+
+;; At this time, the above will return this:
+(SetLink
+   (ListLink
+      (ConceptNode "I")
+      (VariableNode "$star")
+      (ConceptNode "you")
+   )
+   (ListLink
+      (ConceptNode "I")
+      (ConceptNode "love")
+      (VariableNode "$star")
+   )
+)
```

Comment:
Yes, that is the general idea.  The actual code is more like a "proof of concept"; it works in the simplest case and is probably broken in the more complex cases. I have not yet explored enough to even guess what might or might not work, what gotchas there might be in the code.  Yes, it was a Wow! moment for me too, when I first understood the possibilities; so yes, I'm excited by this. But more work is needed.